src/app/components/ConsensusTransactionMethod/index.tsx

```diff
@@ -75,19 +99,93 @@ export const MethodIcon: FC<MethodIconProps> = ({
       >
         {cloneElement(icon, { sx: { fontSize: Math.ceil(size * iconRatio) } })}
       </Box>
-      {label && <Typography sx={{ textTransform: 'capitalize' }}>{label}</Typography>}
+      {label && (
+        <Typography
+          sx={{
+            fontWeight: 'inherit',
+            textTransform: 'capitalize',
+            order: reverseLabel ? -1 : 0,
+            ...(truncate
+              ? {
+                  minWidth: '100px',
+                  maxWidth: '1px',
+                  whiteSpace: 'nowrap',
+                  overflow: 'hidden',
+                  textOverflow: 'ellipsis',
+                }
+              : {}),
+          }}
+        >
+          {label}
+        </Typography>
+      )}
     </Box>
   )
 }
 
-const getConsensusTransactionMethod = (t: TFunction, method: ConsensusTxMethod | undefined) => {
+const MethodIconWithTruncatedLabel: FC<MethodIconProps> = props => {
+  const elementRef = useRef<HTMLDivElement>(null)
+  const [truncate, setTruncate] = useState(false)
+  const [baseLabelWidth, setBaseLabelWidth] = useState(0)
+  const applyTruncate = useCallback(() => {
+    if (elementRef.current) {
+      const elementWidth = elementRef.current.offsetWidth
+      const parentWidth = (elementRef.current.parentNode as HTMLElement)?.offsetWidth
+      setTruncate(elementWidth === parentWidth || baseLabelWidth > parentWidth)
```

Comment:
Why not `elementWidth >= parentWidth`? Why is baseLabelWidth needed / why is elementWidth needed?

Reply:
'>`  elementWidth cannot be larger than parentWidth
if we rely only on elementWidth and parentWidth, element width will be smaller after truncate takes place. so flickering will much more noticable
we cannot rely on baseLabelWidth only because it way be always smaller when initialize on desktop